optimum/intel/openvino/trainer.py

```diff
@@ -536,18 +649,72 @@ def _save(self, output_dir: Optional[str] = None, state_dict=None):
             output_path = os.path.join(output_dir, OV_XML_FILE_NAME)
             self.compression_controller.prepare_for_export()
             model_type = self.model.config.model_type.replace("_", "-")
-            onnx_config_cls = FeaturesManager._SUPPORTED_MODEL_TYPE[model_type][self.feature]
-            onnx_config = onnx_config_cls(self.model.config)
-            use_external_data_format = (
-                onnx_config.use_external_data_format(self.model.num_parameters()) or self.ov_config.save_onnx_model
-            )
-            f = io.BytesIO() if not use_external_data_format else os.path.join(output_dir, "model.onnx")
-            self._onnx_export(self.model, onnx_config, self.ov_config, f)
+            if self.onnx_config is None:
+                onnx_config_cls = FeaturesManager._SUPPORTED_MODEL_TYPE[model_type][self.feature]
+                onnx_config = onnx_config_cls(self.model.config)
+            else:
+                onnx_config = self.onnx_config
+
+            if self._is_pruning_controller_exists():
+                # Note:
+                # OpenVINO provides automated structured pruning of sparse structure in IR.
+                # However it requires static axes, current export utilizes nncf exporter
+                # which generates static-shaped IR.
+
+                f = os.path.join(output_dir, "model.onnx")
+                self.compression_controller.export_model(
```

Comment:
Can we return dynamic shapes after the pruning? It can be easily done using `model.reshape([-1,-1])` after calling `convert_model()`.

Reply:
Collectively decided to retain current implementation for better performance.

Reply:
That is unexpected. All models in optimum-intel are dynamic shapes by default, even though there is better performance for static shapes. Users can easily reshape their model to static shapes for optimal performance (with a possible loss of accuracy resulted from truncating data). Is it just "better performance" or is JQPD not recommended at all for dynamic shapes? If this works with static shapes only it should be very clearly documented.

Reply:
Keep generated IR to static shape until MO/OV runtime is patched.

Reply:
Updates: IR generated by OVTrainer is dynamic now. It is functional for prediction and the model input can be reshaped to static/dynamic shape. Tests on IR for reshape-ability and its equivalence to torch output have been added [6a93a79](https://github.com/huggingface/optimum-intel/pull/150/commits/6a93a79325656fcc3307b3d8759630238d058b87) and [9d30755](https://github.com/huggingface/optimum-intel/pull/150/commits/9d30755d0121e434d258142ade0e4ca71febbcb7)